packages/react-select/src/Select.js

```diff
@@ -415,9 +418,21 @@ export default class Select extends Component<Props, State> {
       const menuOptions = nextProps.menuIsOpen
         ? this.buildMenuOptions(nextProps, selectValue)
         : { render: [], focusable: [] };
-      const focusedValue = this.getNextFocusedValue(selectValue);
-      const focusedOption = this.getNextFocusedOption(menuOptions.focusable);
-      this.setState({ menuOptions, selectValue, focusedOption, focusedValue });
+
+      let focusedValue = null;
+      let focusedOption = null;
+
+      if (nextProps.focusDefaultOption) {
+        focusedValue = this.getNextFocusedValue(selectValue);
+        focusedOption = this.getNextFocusedOption(menuOptions.focusable);
+      }
```

Comment:
A cleaner way to do this would be to check `focusDefaultOption` within `getNextFocusedOption` instead of in this method.

Reply:
And actually, now that I think it about, this logic is incorrect if we don't move it into `this.getNextFocusedOption`.

The situation where this would make a difference is if the menu is open and the `options` prop changes. What `this.getNextFocusedOption` does is it checks to see if the currently selected option is still a focusable option, and if it is, then it keeps it as the focused option (which is still what we want even if `focusDefaultOption` is `false` because there is already a selected option).

So instead we should change `this.getNextFocusedOption` from:
```
getNextFocusedOption(options: OptionsType) {
  const { focusedOption: lastFocusedOption } = this.state;
  return lastFocusedOption && options.indexOf(lastFocusedOption) > -1
    ? lastFocusedOption
    : options[0];
}
```
to:
```
getNextFocusedOption(options: OptionsType, focusDefaultOption: boolean) {
  const { focusedOption: lastFocusedOption } = this.state;
  if (lastFocusedOption && options.indexOf(lastFocusedOption) > -1) {
    return lastFocusedOption;
  }
  return focusDefaultOption ? options[0] : null;
}
```